app/api/access_codes.py

```diff
@@ -118,70 +140,56 @@ def query(self, view_kwargs):
-        """
-        kwargs['user_id'] = current_identity.id
-
-    def before_create_object(self, data, view_kwargs):
-        if view_kwargs.get('ticket_id'):
-            ticket = safe_query(self, Ticket, 'id', view_kwargs['ticket_id'], 'ticket_id')
-            data['event_id'] = ticket.event_id
-        if view_kwargs.get('event_id'):
-            event = safe_query(self, Event, 'id', view_kwargs['event_id'], 'event_id')
-            data['event_id'] = event.id
-        elif view_kwargs.get('event_identifier'):
-            event = safe_query(self, Event, 'identifier', view_kwargs['event_identifier'], 'event_identifier')
-            data['event_id'] = event.id
-        data['user_id'] = current_identity.id
-
-    def after_create_object(self, obj, data, view_kwargs):
-        if view_kwargs.get('ticket_id'):
-            ticket = safe_query(self, Ticket, 'id', view_kwargs['ticket_id'], 'ticket_id')
-            ticket.access_codes.append(obj)
-            self.session.commit()
-
     view_kwargs = True
     decorators = (jwt_required, )
+    methods = ['GET', ]
     schema = AccessCodeSchema
     data_layer = {'session': db.session,
                   'model': AccessCode,
                   'methods': {
                       'query': query,
-                      'before_create_object': before_create_object,
-                      'after_create_object': after_create_object
                   }}
 
 
 class AccessCodeDetail(ResourceDetail):
     """
     AccessCode detail by id
     """
-    def before_get_object(self, view_kwargs):
-        if view_kwargs.get('ticket_id'):
-            ticket = safe_query(self, Ticket, 'id', view_kwargs['ticket_id'], 'ticket_id')
-            if ticket.access_code_id:
-                view_kwargs['id'] = ticket.access_code_id
-            else:
-                view_kwargs['id'] = None
+    def before_update_object(self, access_code, data, view_kwargs):
+        """
+        Method to check for co-organizer access before updating an access code
+        Super-Admin/Admin and Co-organizer can update an access code
+        :param access_code:
+        :param data:
+        :param view_kwargs:
+        :return:
+        """
+        if not has_access('is_coorganizer', event_id=view_kwargs['event_id']):
+            raise ObjectNotFound({'parameter': 'event_id'},
+                                 "Event: {} not found".format(data['event_id']))
+
+    def before_delete_object(self, access_code, view_kwargs):
+        """
+        Method to check for co-organizer access before deleting an access code
+        Super-Admin/Admin can not delete an access code
+        Co-organizer can delete an access code
+        :param access_code:
+        :param view_kwargs:
+        :return:
+        """
+        if has_access('is_admin') and not has_access('is_coorganizer', event_id=view_kwargs['event_id']):
+            raise ForbiddenException({'source': ''}, 'You are not authorized')
 
     decorators = (jwt_required, )
```

Comment:
Why not use `is_coorganizer` for respective methods from the permission manager itself into the decorators